src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs

```diff
@@ -195,7 +195,13 @@ private static void set_machine_sources(ChocolateyConfiguration config, ConfigFi
         private static void set_config_items(ChocolateyConfiguration config, ConfigFileSettings configFileSettings, IFileSystem fileSystem)
         {
             config.CacheLocation = Environment.ExpandEnvironmentVariables(set_config_item(ApplicationParameters.ConfigSettings.CacheLocation, configFileSettings, string.IsNullOrWhiteSpace(configFileSettings.CacheLocation) ? string.Empty : configFileSettings.CacheLocation, "Cache location if not TEMP folder."));
-            if (string.IsNullOrWhiteSpace(config.CacheLocation)) config.CacheLocation = fileSystem.combine_paths(fileSystem.get_temp_path(), "chocolatey"); // System.Environment.GetEnvironmentVariable("TEMP");
+            if (string.IsNullOrWhiteSpace(config.CacheLocation)) {
+                config.CacheLocation = fileSystem.get_temp_path(); // System.Environment.GetEnvironmentVariable("TEMP");
+                if(!config.CacheLocation.EndsWith("chocolatey")) {
+                    config.CacheLocation = fileSystem.combine_paths(fileSystem.get_temp_path(), "chocolatey");

```

Comment:
If `config.CacheLocation` ends with a trailing slash, then `config.CacheLocation.EndsWith("chocolatey")` will not work